Add enterpriseStart MOJO, closes MISC-294, MISC-295, MISC-315

guilgaly opened the pull request:
Allows:
- creating and starting a simulation (or just starting an existing simulation) interactively, `mvn gatling:enterpriseStart`
- starting an existing simulation by calling `enterpriseStart` with a defined `simulationId`, e.g. `mvn gatling:enterpriseStart -Dgatling.enterprise.simulationId`
- creating and starting a simulation without interactive prompts when Maven is run in batch mode, e.g. `mvn --batch-mode gatling:enterpriseStart`

guilgaly commented on pom.xml:
```diff
@@ -84,6 +84,7 @@
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-compat</artifactId>
             <version>${maven3.version}</version>
+            <scope>provided</scope>
```
Unrelated, but `org.apache.maven` plugins should be `provided` (Maven warning message).

slandelle replied:
"Except" that some artifacts actually aren't provided. Warning is too aggressive. I've already done as much clean up as possible.

guilgaly commented on src/main/java/io/gatling/mojo/AbstractEnterprisePluginMojo.java:
```diff
@@ -0,0 +1,119 @@
+  private EnterpriseClient initEnterpriseClient() throws MojoFailureException {
+    if (apiToken == null) {
+      final String msg =
+          "Missing API token\n"
+              + "An API token is required to call the Gatling Enterprise server; see https://gatling.io/docs/enterprise/cloud/reference/admin/api_tokens/ and create a token with the role 'Configure'.\n"
+              + CommonLogMessage.missingConfiguration(
+                  "API token",
+                  "apiToken",
+                  "gatling.enterprise.apiToken",
+                  "GATLING_ENTERPRISE_API_TOKEN",
+                  "MY_API_TOKEN_VALUE");
```
For error messages like this one, the first line (`Missing API token`) isn't very useful when there's a more complete explanation on the second line. However, due to the way Maven displays errors, the beginning of the error message isn't always easy to find/read in the console... So I mostly just wanted to add a line break before any useful information.

guilgaly commented:
Just added two commits:
- `enterpriseUpload` can fallback to the `simulationId` configuration if `packageId` is not defined
- added some Javadoc which is used to generate the help messages (see them with `mvn gatling:help` and `mvn gatling:help -Ddetail=true -Dgoal=<goal name>`
